src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -276,10 +276,21 @@ public override async Task RunAsync()
                 }
             }
 
-            // Check if azure-functions-worker exists in requirements.txt for Python function app
+            // Special checks for python dependencies
             if (workerRuntime == WorkerRuntime.python)
             {
+                // Check if azure-functions-worker exists in requirements.txt for Python function app
                 await PythonHelpers.WarnIfAzureFunctionsWorkerInRequirementsTxt();
+                // Check if remote LinuxFxVersion exists and is different from local version
+                if (!string.IsNullOrEmpty(functionApp.LinuxFxVersion))
+                {
+                    string localImageVersion = await PythonHelpers.ChoosePythonBuildEnvImage();
+                    if (!string.Equals(localImageVersion, functionApp.LinuxFxVersion))
+                    {
+                        string localVersion = (await GetEnvironmentPythonVersion()).Version;
+                        ColoredConsole.WriteLine(WarningColor($"Local python version '{localVersion}' is different from the published version that will be run in Azure on image '{functionApp.LinuxFxVersion}'."));
```

Comment:
nit: might be good to call out that your app might not work in the cloud due to the difference ?  I remember you might get a weird error like "module not found" when deploying 3.6 to 3.7 because the location of the dep folder moved, which makes me wonder if it is best to just fail the operation @Hazhzeng wdyt?

Reply:
Since we allowed customer to publish onto different Python version in old days, and this operation occurs in their local machine, we may not want to completely block this scenario. I think adding one more line in the warning should be sufficient:
`$"This may result in 'ModuleNotFound' error in Azure Functions. Please create a Python {localVersion.Major}.{localVersion.Minor} virtual environment on your local machine for development."`